source/codegen/metadata_mutation.py

```diff
@@ -66,62 +133,68 @@ def mark_coerced_narrow_numeric_parameters(parameters: dict) -> None:
-                    size_param['determine_size_from'] = []
-                size_param['determine_size_from'].append(param['name'])
+                size_param["include_in_proto"] = False
+            if mechanism == "len":
+                if "determine_size_from" not in size_param:
+                    size_param["determine_size_from"] = []
+                size_param["determine_size_from"].append(param["name"])
                 is_optional = common_helpers.is_optional(param)
-                if is_optional != size_param.get('linked_params_are_optional', is_optional):
-                    raise Exception("Code generator does not support linked params that are a mix of optional and required", size_param)
-                size_param['linked_params_are_optional'] = is_optional
+                if is_optional != size_param.get("linked_params_are_optional", is_optional):
+                    raise Exception(
+                        "Code generator does not support linked params that are a mix of optional and required",
+                        size_param,
+                    )
+                size_param["linked_params_are_optional"] = is_optional
+
 
 def get_size_param(param, parameters):
-    named_params = { p['name'] : p for p in parameters }
-    return named_params.get(param['size']['value'], None)
+    named_params = {p["name"]: p for p in parameters}
+    return named_params.get(param["size"]["value"], None)
+
 
 def mark_mapped_enum_params(parameters, enums):
-    for param in (p for p in parameters if 'enum' in p):
-        enum_name = param['enum']
+    for param in (p for p in parameters if "enum" in p):
+        enum_name = param["enum"]
         enum = enums[enum_name]
-        if 'generate-mappings' in enum:
-            del param['enum']
-            param['mapped-enum'] = enum_name
+        if "generate-mappings" in enum:
+            del param["enum"]
+            param["mapped-enum"] = enum_name
 
 
 def populate_grpc_types(parameters, config):
-  for parameter in parameters:
-    if 'callback_params' in parameter:
-        populate_grpc_types(
-            parameter['callback_params'], config)
-    if 'grpc_type' in parameter:
-        continue
-    parameter['grpc_type'] = common_helpers.get_grpc_type(
-        parameter['type'],
-        config)
+    for parameter in parameters:
+        if "callback_params" in parameter:
+            populate_grpc_types(parameter["callback_params"], config)
+        if "grpc_type" in parameter:
+            continue
+        parameter["grpc_type"] = common_helpers.get_grpc_type(parameter["type"], config)
 
 
 def get_short_enum_type_name(typename: str) -> str:
     if typename in ["char[]", "const char[]", "char"]:
         return "String"
     stripped_name = common_helpers.strip_prefix(typename, "Vi")
-    return common_helpers.ensure_pascal_case(stripped_name,)
+    return common_helpers.ensure_pascal_case(
+        stripped_name,
+    )
```

Comment:
This was only split across multiple lines because it had the comma at the end. Remove the comma so it puts it all back on one line.